docs/dev/e2eTesting-GettingStarted.md

```diff
@@ -0,0 +1,96 @@
+# E2E Testing
+## Getting Started
+
+This document describes the steps that need to be taken to run a e2e test from your machine that points to a 
+pre-configured network with a mirror-node.
+
+
+## Setup
+
+Create a configuration file `application.yaml` that has the node configuration, publisher configuration and 
+subscriber configuration along with an operator id and its key.
+
+Example :
+```
+hedera:
+  mirror:
+    monitor:
+      network: PREVIEWNET
+      operator:
+        accountId: 0.0.12345
+        privateKey: <privateKey>
+      publish:
+        clients: 4
+        threads: 40
+        warmupPeriod: 120s
+        scenarios:
+          - name: CryptoTransfer
+            duration: 5m
+            properties:
+              senderAccountId: 0.0.12345
+              recipientAccountId: 0.0.12346
+              amount: 1
+              transferType: CRYPTO
+            tps: 100
+            type: CRYPTO_TRANSFER
+      subscribe:
+        clients: 4
+        grpc:
+        rest:
+          - name: CryptoTransferSub
+            enabled: true
+            samplePercent: 0.05
+```
+
+> The above config targets the `PREVIEWNET` by sending cryptoTransfer transactions for 5 mins with 100 tps
+> using the publisher and then using the subscriber we are sending rest requests to the mirror node to validate these
+> transactions.
```

Comment:
The configuration has 4 clients, do each have 40 threads and is 100 tps for the overall configuration or a single client that we use in our regression tests?

Reply:
No, there's overall 40 publisher threads and 4 sdk clients (each utilizing all nodes). 100 TPS is only for that one publisher scenario.

Reply:
I see, thanks.

Reply:
Dummy questions. What are the publishers? Aren't they the threads in SDK clients?

Reply:
No Leo, A publisher can send multiple transaction types concurrently to the figured network. Please refer [this](https://github.com/hashgraph/hedera-mirror-node/blob/master/docs/monitor.md#publish).

Reply:
So for the configuration mentioned above, the total TPS should be 40 (publisher) x 100 (tps per publisher) = 4000?

Reply:
No, it's 100 TPS total. The clients and threads are just pools that the scenarios use. There's not dedicated threads per client since we're using asynchronous, non-blocking publishers (the SDK itself).